### PR TITLE
module-adapter: Add NULL check for sink buffer

### PR DESCRIPTION
**Pull request by wjablon1**
Just a corner case found in a test with incorrect topology. I think it should be fixed.

Return error if a sink buffer is not present before it is accessed while calculating period size in module_adapter_prepare. This might occur when a pipeline is built incorrectly (without binding) and the sink buffer is not connected to a module.

### CODE REVIEW

File: src/audio/module_adapter/module_adapter.c [hunk 1]
```diff
@@ -460,11 +460,17 @@ int module_adapter_prepare(struct comp_dev *dev)
...
 
-	/* Get period_bytes first on prepare(). At this point it is guaranteed that the stream
-	 * parameter from sink buffer is settled, and still prior to all references to period_bytes.
+	/* Get period_bytes first on prepare(). At this point the stream parameter from sink buffer
+	 * shall be settled, provided that the pipeline is built correctly (bind IPC received).
+	 * Hence check for NULL.
 	 */
 	sink = comp_dev_get_first_data_consumer(dev);
 
+	if (!sink) {
```
Comment (dbaluta) [inline src/audio/module_adapter/module_adapter.c:468]: no need for a new line between call to comp_dev_get_first_data_consumer and sink check.
